lib/src/realtime/src/realtime_channels_interface.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:ably_flutter/src/common/common.dart';
+import 'package:ably_flutter/src/realtime/realtime.dart';
+
+/// A collection of realtime channel objects
+///
+/// https://docs.ably.com/client-lib-development-guide/features/#RTS1
+abstract class RealtimeChannelsInterface<T extends RealtimeChannelInterface>
```

Comment:
I am just moving classes into separate files here. 

There are no interfaces in dart, they are just abstract classes and classes. Unfortunately this (classes which end with "Interface") is a bigger problem than it seems, and would like to handle in a separate PR as part of technical debt in the future.

Reply:
I was thinking it's just a renaming. Is it an agreed convention on Flutter SDKs? A question for "a bigger problem than it seems"

Reply:
I wouldn't say its a Flutter convention. The flutter idea is that every class has an [implicit interface](https://stackoverflow.com/questions/20791286/how-to-define-interfaces-in-dart), you don't need to explicitly have an "interface" keyword. Naming it "Interface" defeats this completely. 

Now, the problem: We currently don't have a `RealtimeChannels`, but we do have both a `RealtimeChannel` and a `RealtimeChannelInterface` (not being used much at all). Confusingly, we also have a `RealtimePlatformsChannels`, which should be renamed to `RealtimeChannels` (it is the implementation of `RealtimeChannels`). It would be inconsistent and confusing to have `RealtimeChannel` be a class, where as `RealtimeChannels` be an interface. I would like to fix this together.

I want to either:
- Remove "interface" from public classes that the user uses. e.g. rename `RealtimeChannelsInterface` to `RealtimeChannels`, and `RealtimeChannelInterface` to `RealtimeChannel`. 
- Delete the interfaces if they're not being used. Some of them really aren't being used 😅. 

Once you digest the above. There might be another problem I want to discuss which I faced a while ago: In some cases, we pass to the user implementation classes, and in other cases, we just pass an abstract class (ending with "Interface"). The interface should be clean to the user: `Fruit`, not `FruitInterface` or `FruitNative`.

Reply:
Another example:

The example app using `ably.Realtime` type, not `ably.RealtimeInterface`. Some of these interfaces are not very useful at all.

Reply:
I've created a PR to address this https://github.com/ably/ably-flutter/pull/193/files, but it is currently in draft.